src/dsp_tools/commands/xmlupload/resource_create_client.py

```diff
@@ -10,15 +14,27 @@ class ResourceCreateClient:
     """client class that creates resources on a DSP server."""
 
     con: Connection
-    project_iri: str
     media_previously_ingested: bool = False
 
     def create_resource(
         self,
-        res_dict: dict[str, Any],
+        graph: Graph,
         resource_has_bitstream: bool,
     ) -> str:
         """Creates a resource on the DSP server, and returns its IRI"""
+        res_dict = _make_json(graph)
         headers = {"X-Asset-Ingested": "true"} if resource_has_bitstream else None
         res = self.con.post(route="/v2/resources", data=res_dict, headers=headers)
         return cast(str, res["@id"])
+
+
+def _make_json(rdf_graph: Graph) -> dict[str, Any]:
```

Comment:
The result of the serialisation is a list. Although the API would accept that, our connection has the payload typed as `dict[str, Any]`. This means that I would either have to change the typing everywhere in the connection (I tried it out, it is not as simple as it sounds), or write a separate post route or not use the connection class. I deem this to be out of scope of this PR therefore, I chose to frame it again. I do this here, because the format of the payload is the business of the API call.